core/domain/story_services.py

```diff
@@ -0,0 +1,318 @@
+    else:
+        story_model = story_models.StoryModel.get(
+            story_id, strict=strict, version=version)
+        if story_model:
+            story = get_story_from_model(story_model)
+            memcache_services.set_multi({story_memcache_key: story})
+            return story
+        else:
+            return None
+
+
+def get_story_summary_by_id(story_id):
+    """Returns a domain object representing a story summary.
+
+    Args:
+        story_id: str. ID of the story summary.
+
+    Returns:
+        StorySummary. The story summary domain object corresponding to
+        a story with the given story_id.
+    """
+    story_summary_model = story_models.StorySummaryModel.get(
+        story_id)
+    if story_summary_model:
+        story_summary = get_story_summary_from_model(
+            story_summary_model)
+        return story_summary
+    else:
+        return None
+
+
+def get_new_story_id():
+    """Returns a new story id.
+
+    Returns:
+        str. A new story id.
+    """
+    return story_models.StoryModel.get_new_id('')
+
+
+def _create_story(committer_id, story, commit_message, commit_cmds):
+    """Creates a new story.
+
+    Args:
+        committer_id: str. ID of the committer.
+        story: Story. story domain object.
+        commit_message: str. A description of changes made to the story.
+        commit_cmds: list(dict). A list of change commands made to the given
+            story.
+    """
+    model = story_models.StoryModel(
+        id=story.id,
+        description=story.description,
+        title=story.title,
+        topic=story.topic,
+        language_code=story.language_code,
+        schema_version=story.schema_version,
+        notes=story.notes,
+        story_contents={
+            'nodes': [
+                story_node.to_dict()
+                for story_node in story.story_nodes
+            ]
+        }
+    )
+    model.commit(committer_id, commit_message, commit_cmds)
+    story.version += 1
+    create_story_summary(story.id)
+
+
+def save_new_story(committer_id, story):
+    """Saves a new story.
+
+    Args:
+        committer_id: str. ID of the committer.
+        story: Story. Story to be saved.
+    """
+    commit_message = (
+        'New story created with title \'%s\'.' % story.title)
+    _create_story(
+        committer_id, story, commit_message, [{
+            'cmd': CMD_CREATE_NEW,
+            'title': story.title
+        }])
+
+
+def compute_summary_of_story(story):
+    """Create a StorySummary domain object for a given Story domain
+    object and return it.
+
+    Args:
+        story_id: str. ID of the story.
+
+    Returns:
+        StorySummary. The computed summary for the given story.
+    """
+    story_model_node_count = len(story.story_nodes)
+    story_summary = story_domain.StorySummary(
+        story.id, story.title, story.language_code,
+        story.version, story_model_node_count,
+        story.topic, story.created_on, story.last_updated
```

Comment:
May need to get topic_id from elsewhere if we're removing it from the Story object.